src/interactions/User/AsEndUser.js

```diff
@@ -51,6 +54,28 @@ export class AsEndUser {
     return this.isLoggedIn() &&
            !!_find(this.user.notifications, {isRead: false})
   }
+
+  /**
+   * Returns a mr-text-* color for the user's displayName
+   */
+  colorCode() {
+    return mapColors(this.user.osmProfile.displayName)
+  }
+}
+
+/**
+ * Returns a mr-text-* color based off a hash of the username.
+ */
+export function mapColors(username) {
+  if (!username) return ""
+
+  return USER_COLOR_MAP[Math.abs(hashCode(username) % 6)]
+}
+
+export function hashCode(s) {
+  for(var i = 0, h = 0; i < s.length; i++)
```

Comment:
Can we use `let` instead of `var`? And if you happen to remember where you got this from, can you add a comment with a link to it?